core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1458,9 +1458,78 @@ class KafkaApis(val requestChannel: RequestChannel,
   }
 
   def handleTxnOffsetCommitRequest(request: RequestChannel.Request): Unit = {
```

Comment:
You are correct. There are subtle differences in minor things like log messages. I will look into the best way to consolidate.